x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/pivot/AggregationResultUtils.java

```diff
@@ -97,4 +97,35 @@
         });
     }
 
+    @SuppressWarnings("unchecked")
+    static void updateDocument(Map<String, Object> document, String fieldName, Object value) {
+        String[] fieldTokens = fieldName.split("\\.");
```

Comment:
in 99% of cases `fieldName` is flat, so I wonder if we should have a shortcut optimization like:

```
if (fieldName.contains(".")) {
    document.put(fieldName, value);
    return;
}
```

and then briefly explain as a code comment what the special handling does.

Reply:
That sounds good to me. It may provide a logic flow optimization, but I doubt there will be a performance one as `updateDocument` will just add the field and return.

Reply:
Looking at how `String#split` is written, it does a handful of constant time calculations, and since we are splitting on a single char `.`, `String#split` when the char does not exist in the string is essentially the same as doing `String.contains` once. 

I will add a clause that checks `if (fieldTokens.length == 1)` as an early shortcut, but I don't think it will cause any performance increase.